src/Compiler/Facilities/DiagnosticOptions.fs

```diff
@@ -31,3 +31,6 @@ type FSharpDiagnosticOptions =
             WarnAsError = []
             WarnAsWarn = []
         }
+
+    member x.CheckXmlDocs =
+        List.contains 3390 x.WarnOn && not (List.contains 3390 x.WarnOff)
```

Comment:
Is this correct? [Docs for `warnon`](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-options) say "Enable specific warnings that might be off by default or **disabled by another command-line option**." So should we care about if it's in the `WarnOff` list?

Reply:
Currently, for warnings, it checks if they are not disabled even when they are enabled
https://github.com/dotnet/fsharp/blob/cecc728fb8aadc41647c24daed34a59186225acb/src/Compiler/Driver/CompilerDiagnostics.fs#L396-L400


However, if the warning is enabled and added to WarnAsError, then the rules are somewhat complicated, and they should also be taken into account, hm.
https://github.com/dotnet/fsharp/blob/cecc728fb8aadc41647c24daed34a59186225acb/src/Compiler/Driver/CompilerDiagnostics.fs#L419-L422